app/controllers/missions_controller.rb

```diff
@@ -1,47 +1,49 @@
 class MissionsController < ApplicationController
-  VALID_MISSION_PARAMS = %w(name description)
+  VALID_MISSION_PARAMS = %w(name description public)
 
-  before_action :set_mission, only: [:show, :edit, :update, :destroy]
+  before_action(:set_mission, {
+    only: [:show, :edit, :update, :destroy]
+  })
+
+  before_action({ only: [:show, :edit] }) do
+    require_user! unless @mission.public?
+  end
 
-  # GET /missions
-  # GET /missions.json
   def index
-    @missions = Mission.all
+    @missions = Mission.where({ public: true })
```

Comment:
It won't, the idea was that you'd have your missions on either a) an activity page (like github) or b) a nested route like `users/{me}/missions` -- think this will work?